db/blob_log_format.h

```diff
@@ -0,0 +1,202 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+//
+// Log format information shared by reader and writer.
+
+#pragma once
+
+#include <cstdint>
+#include <utility>
+#include <string>
+#include <cstddef>
+#include "rocksdb/status.h"
+#include "rocksdb/types.h"
+
+namespace rocksdb {
+
+  class BlobFile;
+
+namespace blob_log {
+
+enum RecordType {
+  // Zero is reserved for preallocated files
+  kFullType = 0,
+
+  // For fragments
+  kFirstType = 1,
+  kMiddleType = 2,
+  kLastType = 3,
+};
+
+enum RecordSubType {
+
+  kRegularType = 0,
+  kTTLType = 1,
+  kTimestampType = 2,
+};
+
+const uint32_t kMagicNumber = 2395959;
+
+class Reader;
+
+class BlobLogHeader {
+
+ private:
+
+   uint32_t magic_number_ = 0;
+   std::pair<uint32_t, uint32_t> ttl_guess_;
+   std::pair<uint64_t, uint64_t> ts_guess_;
+   bool has_ttl_;
+   bool has_ts_;
+
+ public:
+
+  // magic number + flags + ttl guess + timestamp range
+  static const size_t kHeaderSize = 4 + 4 + 4 * 2 + 8 * 2;
+  // 32
+
+  void setTTL(bool ttl = true)  { has_ttl_ = ttl; }
+
+  void setTimestamps(bool ts = true) { has_ts_ = ts; }
+
+  void setTTLGuess(const std::pair<uint32_t, uint32_t>& ttl) { ttl_guess_ = ttl; has_ttl_ = true; }
```

Comment:
We use 80-char wrapping rule. Run "make format" will fix many of those problems automatically.